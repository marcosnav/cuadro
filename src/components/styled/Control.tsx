import styled from 'styled-components';

interface IProps {
  disabled?: boolean;
}

const Control = styled.button<IProps>`
  align-items: center;
  background: ${(props) => props.theme.WHITE};
  border: none;
  color: ${(props) => props.theme.DARK};
  display: flex;
  flex-direction: column;
  font-size: 0.64em;
  font-weight: bold;
  height 42px;
  justify-content: space-between;
  opacity: ${(props) => props.disabled ? '0.3' : '1'};
  padding: 0;
  transition: opacity 0.3s ease;

  &:focus {
    outline: none;
  }
`;

Control.displayName = 'Control';

export default Control;
