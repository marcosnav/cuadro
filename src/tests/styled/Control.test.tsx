import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import Control from './../../components/styled/Control';

describe('<Control/>', () => {
  test('renders properly', () => {
    const component = mount(<Control>Control text</Control>);
    expect(component.text()).toBe('Control text');
    expect(component).toMatchSnapshot();
  });

  describe('when disabled', () => {
    const component = mount(<Control disabled={true} >disabled</Control>);

    test('must be opaque', () => {
      expect(component).toHaveStyleRule('opacity', '0.3');
    });

    test('when re-enabled, click must work and be full color', () => {
      component.setProps({ disabled: false });
      expect(component).toHaveStyleRule('opacity', '1');
    });
  });
});
