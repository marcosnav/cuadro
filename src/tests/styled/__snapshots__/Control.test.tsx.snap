// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Control/> renders properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.64em;
  font-weight: bold;
  height: 42px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 0;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
}

.c0:focus {
  outline: none;
}

<Control>
  <button
    className="c0"
  >
    Control text
  </button>
</Control>
`;
